deploy/install/01-prerequisite/01-namespace.yaml

```diff
@@ -2,3 +2,10 @@ apiVersion: v1
 kind: Namespace
 metadata:
   name: longhorn-system
+  labels:
+    pod-security.kubernetes.io/enforce: privileged
+    pod-security.kubernetes.io/enforce-version: latest
+    pod-security.kubernetes.io/audit: privileged
+    pod-security.kubernetes.io/audit-version: latest
+    pod-security.kubernetes.io/warn: privileged
+    pod-security.kubernetes.io/warn-version: latest
```

Comment:
1. For the pods that don't satisfy the `privileged` policy requirements, will Kubernetes block them or just generate warnings/audit annotations?
2. Why is the (minor) version `latest` rather than a specific version like v1.25?

Reply:
> 1. For the pods that don't satisfy the privileged policy requirements, will Kubernetes block them or just generate warnings/audit annotations?

With this setting, Kubernetes doesn't block/warn any pods (privileged/baseline/restricted) .

* `pod-security.kubernetes.io/enforce: privileged` means blocking pods requiring more permission than `privileged`. That means don't block any pods inside `longhorn-system` namespace because there is no pod that requires more permission than the privileged level.
* `pod-security.kubernetes.io/audit: privileged` means generating audit events for pods requiring more permission than `privileged`. That means don't generate any events for pods inside `longhorn-system` namespace because there is no pod that requires more permission than the privileged level.
* `pod-security.kubernetes.io/audit: privileged` means generating warning message for pods requiring more permission than `privileged`. That means don't generate any warning for pods inside `longhorn-system` namespace because there is no pod that requires more permission than the privileged level.

> Why is the (minor) version latest rather than a specific version like v1.25?

It doesn't really matter which Pod Security Standard version we are pinning here as the setting we are doing is `don't block/warn any pod according to xyz PSA version`. This will instruct the PSA controller not to use the default value in case the default value is `blocking if missing labels on the namespace`.

Reply:
Got it. I was thinking `enforce` will block the unsatisfied pods while `audit` or `warn` will allow the pods, which is a conflict. But actually, the latter two are just for warning or audit rather than accepting or rejecting the pods.

As for the versioning, the doc does not explain too much about this part. If the version is `latest`, does this mean that the PSS we add applies to the Longhorn namespace under any Kubernetes version?

Reply:
> As for the versioning, the doc does not explain too much about this part. If the version is latest, does this mean that the PSS we add applies to the Longhorn namespace under any Kubernetes version?

There is an example at [this link](https://kubernetes.io/docs/setup/best-practices/enforcing-pod-security-standards/#adopt-a-multi-mode-strategy)

I think the PSA controller will:
1. Get the current latest Kubernetes version from upstream and get the PSS definitions from that version
2. Use these definitions to evaluate pods